src/main/java/de/tum/www1/orion/build/instructor/OrionInstructorBuildUtil.kt

```diff
@@ -35,26 +35,22 @@ interface CustomizableCheckoutPath {
 enum class RepositoryCheckoutPath : CustomizableCheckoutPath {
     ASSIGNMENT {
         override fun forProgrammingLanguage(language: ProgrammingLanguage): String {
-            return when (language) {
-                ProgrammingLanguage.JAVA, ProgrammingLanguage.PYTHON, ProgrammingLanguage.C -> "assignment"
-                else -> throw IllegalArgumentException("Repository checkout path for assignment repo has not yet been defined for $language")
```

Comment:
The when expression in kotlin, when used with enum or sealed class, is exhaustive. So that code path is not reachable. If someone adds a new value into the enum, the handler for the new value will be enforced by the compiler so they can't forget it.